tracer/test/Datadog.Trace.Tests/MultipartFormTests.cs

```diff
@@ -0,0 +1,111 @@
+                requestBody = sreader.ReadToEnd();
+            };
+
+            var url = new Uri($"http://localhost:{agent.Port}/");
+            var factory = new ApiWebRequestFactory(url, AgentHttpHeaderNames.DefaultHeaders);
+            var request = (IMultipartApiRequest)factory.Create(url);
+            await request.PostAsync(new MultipartFormItem[]
+            {
+                new("Name 1", MimeTypes.Json, "FileName 1.json", new ArraySegment<byte>(new byte[] { 42 })),
+                new("Name 2", MimeTypes.MsgPack, "FileName 2.msgpack", new ArraySegment<byte>(new byte[] { 42 })),
+            });
+
+            Assert.NotNull(requestBody);
+            await Verifier.Verify(requestBody);
+        }
+
+        [Fact]
+        public async Task ApiWebRequestValidationTest()
+        {
+            using var agent = (MockTracerAgent.TcpUdpAgent)EnvironmentHelper.GetMockAgent();
+            agent.ShouldDeserializeTraces = false;
+            string requestBody = null;
+            agent.RequestReceived += (sender, args) =>
+            {
+                var ctx = args.Value;
+                var rq = ctx.Request;
+                using var sreader = new StreamReader(rq.InputStream, Encoding.ASCII);
+                requestBody = sreader.ReadToEnd();
+            };
+
+            var url = new Uri($"http://localhost:{agent.Port}/");
+            var factory = new ApiWebRequestFactory(url, AgentHttpHeaderNames.DefaultHeaders);
+            var request = (IMultipartApiRequest)factory.Create(url);
+            await request.PostAsync(new MultipartFormItem[]
+            {
+                new("Name\" 1\"", MimeTypes.Json, "FileName 1.json", new ArraySegment<byte>(new byte[] { 42 })),
+                new("Name 2", MimeTypes.MsgPack, "FileName '2'.msgpack", new ArraySegment<byte>(new byte[] { 42 })),
+            });
+
+            Assert.Equal(string.Empty, requestBody);
+        }
+#if NETCOREAPP3_1_OR_GREATER
+
+        [Fact]
+        public async Task HttpClientRequestMultipartTest()
+        {
+            using var agent = (MockTracerAgent.TcpUdpAgent)EnvironmentHelper.GetMockAgent();
+            agent.ShouldDeserializeTraces = false;
+            string requestBody = null;
+            agent.RequestReceived += (sender, args) =>
+            {
+                var ctx = args.Value;
+                var rq = ctx.Request;
+                using var sreader = new StreamReader(rq.InputStream, Encoding.ASCII);
+                requestBody = sreader.ReadToEnd();
+            };
+
+            var url = new Uri($"http://localhost:{agent.Port}/");
+            var factory = new HttpClientRequestFactory(url, AgentHttpHeaderNames.DefaultHeaders);
+            var request = (IMultipartApiRequest)factory.Create(url);
+            await request.PostAsync(new MultipartFormItem[]
+            {
+                new("Name 1", MimeTypes.Json, "FileName 1.json", new ArraySegment<byte>(new byte[] { 42 })),
```

Comment:
I'd also be interested in a file name without spaces to see if HttpClient generates both `filename` and `filename*=utf-8''`

Reply:
Yes, I've already tested that scenario, the HttpClient always generates the utf8 field. I didn't implement it for ApiWebRequest because is not required ( is not in the original RFC ) and because the usage is going to be internal I'm not worried about it.

That's the reason I've added the warning: https://github.com/DataDog/dd-trace-dotnet/blob/4b08f0499a40d8fa58c075f65d406b08f772695c/tracer/src/Datadog.Trace/Agent/Transports/ApiWebRequest.cs#L61